h2o-algos/src/main/java/hex/tree/FriedmanPopescusH.java

```diff
@@ -216,9 +231,12 @@ static Frame computeFValues(int[] modelIds, Frame filteredFrame, String[] cols,
         Frame uniqueWithCounts = uniqueRowsWithCounts(filteredFrame);
         Frame uncenteredFvalues = new Frame(partialDependence(modelIds, uniqueWithCounts, learnRate, sharedTreeSubgraphs).vec(0));
         VecUtils.DotProduct multiply = new VecUtils.DotProduct().doAll(uniqueWithCounts.vec("nrow"), uncenteredFvalues.vec(0));
-        double meanUncenteredFValue = multiply.result / filteredFrame.numRows();
-        for (int i = 0; i < uncenteredFvalues.numRows(); i++) {
-            uncenteredFvalues.vec(0).set(i, uncenteredFvalues.vec(0).at(i) - meanUncenteredFValue);
+        final double meanUncenteredFValue = multiply.result / filteredFrame.numRows();
+        try (Vec.Writer uncenteredFValuesWriter = uncenteredFvalues.vec(0).open()) {
+            Vec.Reader uncenteredFValuesReader = uncenteredFvalues.vec(0).new Reader();
+            for (int i = 0; i < uncenteredFvalues.numRows(); i++) {
+                uncenteredFValuesWriter.set(i, uncenteredFValuesReader.at(i) - meanUncenteredFValue);
```

Comment:
Here I use `Vec.Reader` and `Vec.Writer`. Both are holding the same chunk in cache but since we are not reading after writing the same entry and we are not using rollup stats here, this should be safe.

`Vec.Reader` holds a cached chunk so we don't have to call `chunkForRow` on every access but it gets run once we run out of rows in the cached chunk to get a next chunk.